src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -173,13 +173,13 @@ public async Task<string> GetLineTextAfterCaretAsync(CancellationToken cancellat
             return lineTextAfterCaret;
         }
 
-        public async Task MoveCaretAsync(int position, CancellationToken cancellationToken)
+        public async Task MoveCaretAsync(int position, CancellationToken cancellationToken, string contentType = ContentTypeNames.RoslynContentType)
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             var view = await GetActiveTextViewAsync(cancellationToken);
 
-            var subjectBuffer = view.GetBufferContainingCaret();
+            var subjectBuffer = view.GetBufferContainingCaret(contentType);
             Assumes.Present(subjectBuffer);
 
             var point = new SnapshotPoint(subjectBuffer.CurrentSnapshot, position);
```

Comment:
This seems very very broken -- the caret position here is implicitly being treated as a view buffer position, but we're converting it to the subject buffer like this which could be entirely broken. I'm not sure what this is trying to do, but rather than having to pass around content type we can probably just delete this...?

Chasing this through source control it seems to be _very_ old; I'm guessing we never noticed if we didn't actually have integration tests for ASP.NET cases.

Reply:
It looked funky to me as well, but I was kind of scared to change anything here as I don't have much expertise. Just to confirm, you are recommending that we avoid converting to subject buffer and directly invoke the MoveTo based off the view, like below: https://github.com/dotnet/roslyn/blob/c44c7e89a8cab8c9862f710b62a745daf0e26c9a/src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs#L987